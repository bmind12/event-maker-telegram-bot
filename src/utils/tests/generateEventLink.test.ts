import {
    CalendarEvent,
    CalendarType,
    generateEventLink,
} from '../generateEventLink';

describe('generateEventLink', () => {
    // UTC is used to be indifferent to locales
    const startDate = new Date(Date.UTC(2020, 11, 20, 16, 30, 0));
    const endDate = new Date(Date.UTC(2020, 11, 21, 1, 0, 0));
    const event: CalendarEvent = {
        title: 'New year party',
        location: 'Politických vězňů 1511/5',
        allDay: false,
        start: {
            day: startDate.getDate(),
            month: startDate.getMonth() + 1, // months in JS start with 0
            year: startDate.getFullYear(),
            optional: {
                hours: startDate.getHours(),
                minutes: startDate.getMinutes(),
            },
        },
        end: {
            day: endDate.getDate(),
            month: endDate.getMonth() + 1, // months in JS start with 0
            year: endDate.getFullYear(),
            optional: {
                hours: endDate.getHours(),
                minutes: endDate.getMinutes(),
            },
        },
    };

    it('generates Google link', () => {
        expect(generateEventLink(event, CalendarType.Google)).toMatchSnapshot();
    });

    it('generates ICS string', () => {
        expect(generateEventLink(event, CalendarType.ICS)).toMatchSnapshot();
    });

    it('generates Outlook link', () => {
        expect(
            generateEventLink(event, CalendarType.Outlook)
        ).toMatchSnapshot();
    });
});
