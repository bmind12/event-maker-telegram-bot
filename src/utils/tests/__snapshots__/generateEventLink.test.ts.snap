// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateEventLink generates Google link 1`] = `"https://calendar.google.com/calendar/render?action=TEMPLATE&dates=20201220T163000Z%2F20201221T010000Z&location=Politick%C3%BDch%20v%C4%9Bz%C5%88%C5%AF%201511%2F5&text=New%20year%20party"`;

exports[`generateEventLink generates ICS string 1`] = `
"BEGIN:VCALENDAR
VERSION:2.0
BEGIN:VEVENT
DTSTART:20201220T163000Z
DTEND:20201221T010000Z
SUMMARY:New year party
LOCATION:Politických vězňů 1511/5
END:VEVENT
END:VCALENDAR
"
`;

exports[`generateEventLink generates Outlook link 1`] = `"https://outlook.live.com/calendar/0/deeplink/compose?enddt=2020-12-21T01:00:00&location=Politick%C3%BDch%20v%C4%9Bz%C5%88%C5%AF%201511%2F5&path=%2Fcalendar%2Faction%2Fcompose&rru=addevent&startdt=2020-12-20T16:30:00&subject=New%20year%20party"`;
